internal/pkg/diff/reporter.go

```diff
@@ -130,10 +171,91 @@ func pathToString(path cmp.Path) string {
 		case cmp.MapIndex:
 			parts = append(parts, cast.ToString(v.Key().Interface()))
 		case cmp.SliceIndex:
-			parts = append(parts, cast.ToString(v.Key()))
+			// index1 or index2 can be "-1",
+			// if the value is on one side only
+			index1, index2 := v.SplitKeys()
+			parts = append(parts, cast.ToString(math.Max(float64(index1), float64(index2))))
 		case cmp.StructField:
 			parts = append(parts, v.Name())
 		}
 	}
+
 	return strings.Join(parts, ".")
 }
+
+func (r *Reporter) objectPath(value reflect.Value) string {
+	if value.IsValid() {
+		if v, ok := value.Interface().(model.RecordPaths); ok {
```

Comment:
Ak ma value `Path` metodu, tak ju pouzijeme pri vypise, tyka sa to transformation `Block`, pouzite v dalsom PR.